pages/live/lobby.js

```diff
@@ -207,26 +275,26 @@ async function lobbyInit () {
 
       // Fetch the api to change the lobby name
       const request = await fetch(`/api/lobbies/rename/${encodedName}/${newName}`);
-      if (request.status !== 200) {
+      let requestData;
+      try {
+        requestData = await request.json();
+      } catch (e) {
         return showPopup("Unknown error", "The server returned an unexpected response. Error code: " + request.status, POPUP_ERROR);
-      } else {
-        const data = await request.json();
-        switch (data) {
-
-          case "SUCCESS":
-            showPopup("Success", "The lobby name has been successfully updated.");
-            return;
+      }
 
-          case "ERR_LOGIN": return showPopup("Not logged in", "Please log in via Steam before editing lobby details.", POPUP_ERROR);
-          case "ERR_STEAMID": return showPopup("Unrecognized user", "Your SteamID is not present in the users database. WTF?", POPUP_ERROR);
-          case "ERR_NAME": return showPopup("Lobby not found", "An open lobby with this name does not exist.", POPUP_ERROR);
-          case "ERR_NEWNAME": return showPopup("Invalid lobby name", "Please keep the lobby name to 50 characters or less.", POPUP_ERROR);
-          case "ERR_EXISTS": return showPopup("Lobby name taken", "A lobby with this name already exists.", POPUP_ERROR);
-          case "ERR_PERMS": return showPopup("Permission denied", "You do not have permission to perform this action.", POPUP_ERROR);
+      switch (requestData) {
+        case "SUCCESS":
+          showPopup("Success", "The lobby name has been successfully updated.");
+          return;
 
-          default: return showPopup("Unknown error", "The server returned an unexpected response: " + data, POPUP_ERROR);
+        case "ERR_LOGIN": return showPopup("Not logged in", "Please log in via Steam before editing lobby details.", POPUP_ERROR);
+        case "ERR_STEAMID": return showPopup("Unrecognized user", "Your SteamID is not present in the users database. WTF?", POPUP_ERROR);
+        case "ERR_NAME": return showPopup("Lobby not found", "An open lobby with this name does not exist.", POPUP_ERROR);
+        case "ERR_NEWNAME": return showPopup("Invalid lobby name", "Please keep the lobby name to 50 characters or less.", POPUP_ERROR);
+        case "ERR_EXISTS": return showPopup("Lobby name taken", "A lobby with this name already exists.", POPUP_ERROR);
+        case "ERR_PERMS": return showPopup("Permission denied", "You do not have permission to perform this action.", POPUP_ERROR);
 
-        }
+        default: return showPopup("Unknown error", "The server returned an unexpected response: " + requestData, POPUP_ERROR);
```

Comment:
This particular (or _very_ similar) switch statement(s) gets repeated _a lot_. Consider splitting this into its own function.

Reply:
Considered, but for a bigger frontend refactor. API queries like these are everywhere, and the handling is different in every file. Should probably be a part of `tools.js`. My point is, it's outside of the scope for this PR.

Reply:
Understandable, but a temporary function would still be cleaner than repetition imo